### PR TITLE
Typing investigation

### PR DESCRIPTION
**Pull request by InvincibleRMC**
Ben loves all my PR's so much

### CODE REVIEW

File: surface/gui/gui/app.py [hunk 1]
```diff
@@ -31,7 +32,7 @@ def __init__(self):
         self.logger: Logger = Logger()
         layout.addWidget(self.logger, 1, 0)
 
-    def closeEvent(self, event):
+    def closeEvent(self, a0: QCloseEvent):
```
Comment (benjaminwp18) [inline surface/gui/gui/app.py:35]: `a0`?

File: surface/gui/gui/event_nodes/server.py [hunk 2]
```diff
@@ -19,15 +19,11 @@ def __init__(self, interface: type, topic: str, callback: callable):
...
-        """Spin the server in a new thread."""
-        # Create new executor to avoid spinning global executor multiple times
-        executor = rclpy.executors.SingleThreadedExecutor()
+        executor = SingleThreadedExecutor()
         executor.add_node(self)
-        Thread(target=self.executor.spin, daemon=True,
+        Thread(target=executor.spin, daemon=True,
                name=f'{self.node_name}_spin').start()
 
-        self.executor = executor
```
Comment (benjaminwp18) [inline surface/gui/gui/event_nodes/server.py:30]: Removing this leaves executor as a local var, meaning `kill_executor` can't find it
Maybe just declare the executor as self.executor from the beginning?

File: surface/gui/gui/event_nodes/server.py [hunk 3]
```diff
@@ -19,15 +19,11 @@ def __init__(self, interface: type, topic: str, callback: callable):
 
         self.srv = self.create_service(interface, topic, callback)
 
-    def spin_async(self):
```
Comment (benjaminwp18) [inline surface/gui/gui/event_nodes/server.py:22]: I originally did the `spin_async`s for parity with `publish` and `send_request_async` in the publisher and client event nodes; that way for every event node you call the constructor to make the node, then call the specific communication function to communicate. Thoughts?

File: surface/gui/gui/event_nodes/subscriber.py [hunk 4]
```diff
@@ -2,33 +2,27 @@
...
 
-    def spin_async(self):
-        """Wait for message in a different thread, emit signal with message when received."""
-        executor = rclpy.executors.SingleThreadedExecutor()
+        executor = SingleThreadedExecutor()
         executor.add_node(self)
         Thread(target=executor.spin, daemon=True,
                name=f'{self.node_name}_spin').start()
 
-        self.executor = executor
```
Comment (benjaminwp18) [inline surface/gui/gui/event_nodes/subscriber.py:31]: I don't think `kill_executor` can see this executor now, either

File: surface/gui/gui/modules/logger.py [hunk 5]
```diff
@@ -10,32 +10,14 @@
 from gui.event_nodes.subscriber import GUIEventSubscriber
 
 # Names and log text colors for each message severity
-SEVERITY_LEVELS = [
```
Comment (benjaminwp18) [inline surface/gui/gui/modules/logger.py:13]: agghhh my javascript brain made this monstrosity
Thank you for obliterating it

File: surface/gui/gui/modules/logger.py [hunk 6]
```diff
@@ -10,32 +10,14 @@
...
-    }
-]
+SEVERITY_LEVELS_NAMES = ['Unset', 'Debug', 'Info', 'Warn', 'Error', 'Fatal']
+SEVERITY_LEVELS_QCOLORS = [QColor(0, 0, 0),
+                           QColor(50, 50, 50),
+                           QColor(150, 150, 150),
+                           QColor(150, 150, 0),
+                           QColor(255, 0, 0),
+                           QColor(168, 0, 0)
+                           ]
```
Comment (benjaminwp18) [inline surface/gui/gui/modules/logger.py:20]: You have the weirdest dangling bracket style I have ever seen

File: surface/task_selector/task_selector/manual_control_node.py [hunk 7]
```diff
@@ -1,91 +1,91 @@
...
+            rov_msg.roll = float(-buttons[L1] + buttons[R1])
             self.pixhawk_publisher.publish(rov_msg)
 
     # Used to create smoother adjustments
     def joystick_profiles(self, val: float):
         return val * abs(val)
 
-    # TODO what is a goal_handle????
-    def execute_callback(self, goal_handle):
+    def execute_callback(self, goal_handle: ServerGoalHandle) -> BasicTask.Result:
```
Comment (benjaminwp18) [inline surface/task_selector/task_selector/manual_control_node.py:88]: It's beautiful

File: surface/task_selector/task_selector/task_requestor.py [hunk 8]
```diff
@@ -50,19 +50,22 @@ def request_task_callback(self, request: TaskRequest.Request, response: TaskRequ
...
-        elif request.task_id == Tasks.EX_GOOD_MORNING.value:
-            self.send_morning_goal(True, True)
-        elif request.task_id == Tasks.EX_TIMED.value:
-            self.send_basic_goal(self.timed_task_client)
-        elif request.task_id == Tasks.EX_BASIC.value:
-            self.send_basic_goal(self.basic_task_client)
-        elif request.task_id == Tasks.CANCEL.value:
-            response.response = "Canceled"
+        task_id = request.task_id
+        if isinstance(task_id, int):
```
Comment (benjaminwp18) [inline surface/task_selector/task_selector/task_requestor.py:54]: liberatore says this is an excellent opportunity to exit early:
```python
if not isinstance(task_id, int):
     self.get_logger().fatal("task_id is not int!")
     return response
```

Also, do we need type checking coming out of a ROS interface? If we don't interpret those types as guaranteed, that's gonna mean a lot of `isinstance`

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

File: surface/gui/gui/app.py (hunk 1 above)
Comment (InvincibleRMC) [reply surface/gui/gui/app.py:35]: When overloading a method you should use what is was called. PyQT5 called it an a0

File: surface/gui/gui/event_nodes/server.py (hunk 3 above)
Comment (InvincibleRMC) [reply surface/gui/gui/event_nodes/server.py:22]: You want to actively make people publish and send requests
But for listeners its just asking for people to forget to call spin for no reason

File: surface/gui/gui/app.py (hunk 1 above)
Comment (benjaminwp18) [reply surface/gui/gui/app.py:36]: Add a comment so I don't change it

File: surface/task_selector/task_selector/manual_control_node.py [hunk 9]
```diff
@@ -1,117 +1,116 @@
...
-    DPADVERT:        int = 7
+class ManualControlNode(Node):
+    _passing: bool = False
 
     def __init__(self):
         super().__init__('manual_control_node',
                          parameter_overrides=[])
-
-        self._action_server = ActionServer(
+        # TODO would Service make more sense then Actions?
```
Comment (benjaminwp18) [inline surface/task_selector/task_selector/manual_control_node.py:44]: No; we need feedback over time
Remove this

**Review by benjaminwp18 — CHANGES_REQUESTED** [review]

**Review by benjaminwp18 — APPROVED** [review]

**Pull request merged**